src/Microsoft.Health.Fhir.Core/Features/Persistence/ResourceDeserializer.cs

```diff
@@ -38,6 +40,88 @@ public ResourceElement Deserialize(ResourceWrapper resourceWrapper)
             return resource;
         }
 
+        /// <summary>
+        /// Deserialize RawResource in ResourceWrapper to a JsonDocument.
+        /// If the RawResource's VersionSet or LastUpdatedSet are false, then the RawResource's data will be updated
+        /// to have them set to the values in the ResourceWrapper
+        /// </summary>
+        /// <param name="resourceWrapper">Input ResourceWrapper to convert to a JsonDocument</param>
+        /// <returns>A <see cref="JsonDocument"/></returns>
+        public static JsonDocument DeserializeToJsonDocument(ResourceWrapper resourceWrapper)
+        {
+            EnsureArg.IsNotNull(resourceWrapper, nameof(resourceWrapper));
+
+            if (resourceWrapper.RawResource.LastUpdatedSet && resourceWrapper.RawResource.VersionSet)
```

Comment:
Is it a valid assumption that a resource is guaranteed to have versionId/lastUpdated if it was created/modified after some timestamp T (can be in near future)? If yes, is it possible/acceptable to use T (global) for decision here instead of per document flags?

Reply:
That's an interesting idea. But then we would need to block writes from an earlier version of the code after that time.

Reply:
Right, we can select T with enough buffer to account for deployment/rollback.

Reply:
Keep in mind that we are not the only ones running this as a service. So such a solution would need to be more deterministic, where we we block usage based on an earlier version of the code

Reply:
I don't think this works, as users can do a create using a PUT. With these changes, we only set the flags to true if the create came in as a POST. On a PUT, we don't know if it will be a create or update (and thus the version ID) until the call comes back from the data layer to Cosmos/SQL.

Reply:
Discussed with Yazan offline. Though we can make it work in case of PUT (needs extra trip to DB to fetch previous version id), plan is to not do it for now to avoid any PUT perf regression (with the understanding that future reads of this resource will have deser/ser cost).

Reply:
BTW, can we use persisted request method (PUT vs POST) along with timestamp T to differentiate resources with/without versionId?
method == POST and lastUpdated > T  => avoid deser/ser

Reply:
That could work too, but we'd have to store the date that the FHIR server was upgraded to the version that has these changes as the T value. That could get complicated when you have multiple compute deployments as you should only have T stored once all the compute instances have been upgraded to the this version of the code.

Reply:
Consider T is coming from configs (with default value = max timestamp, which means optimization is disabled). Now when all the clusters are upgraded, we can apply config change (T = current timestamp) to the system (this can be in the next release). Does that make sense? 
All this will help in avoiding schema migration complexities and increase in data size per document. Right?